### PR TITLE
Remove ability for AI Eye to swap borg cells

### LINKED ISSUE #15851: AI eye can swap borg power cells out in dock
### Labels

[BUG][MINOR]

### Description

The AI can swap out power cells of borgs in charging docks while in the intangible AI eye

### Steps to reproduce

1.  be ai eye
2. see borg in charging dock
3. swap their cell
4. happy borg

### Expected Behavior

Not being able to swap cells as an AI GHOST. 

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by MeggalBozale**
## About the PR 
Intangible AI eyes can no longer swap cells for cyborgs in docking stations.

## Why's this needed? 
Fixes #15851 by ensuring cells swappers are not intangble.

### CODE REVIEW

**Review by pgmzeta — CHANGES_REQUESTED** [review]
please do the same check for `cell-remove` or AIs can brick cyborgs and not be able to fix them

File: code/modules/robotics/robot/robot_docking_station.dm [hunk 1]
```diff
@@ -977,6 +977,9 @@ TYPEINFO(/obj/machinery/recharge_station)
 			if (user == src.occupant)
 				boutput(user, "<span class='alert'>You can't modify your own power cell!</span>")
 				return
+			if (isintangible(user))
```
[the remark this answers is not in the document]
Comment (MeggalBozale) [reply code/modules/robotics/robot/robot_docking_station.dm:980]: Might catch other intangible, devious spirits, but I have no issue with it being isAIeye

**Review by pgmzeta — APPROVED** [review]

**Comment by 444Portal** [comment]
Not entirely sure why this is a bug that needs fixed - If the issue is the AI Eye ghost is swapping it, and less about the AI doing it, then, it's fine, right? As Docking Stations seem to be somewhat automated? I mean, how does a Roboticist Swap Cells when the sprite shoes an active Docking Station as shut? Dunno.

**Comment by MeggalBozale** [comment]
I always imagine there's some physical part to cell swapping in the station that doesn't show visually.

**Comment by Tarmunora** [comment]
🤷 if the idea is to make cell swapping an action for humans to perform instead of the AI/etc. wouldn't checking for silicon be better? Its not like the AI can't just deploy an eyebot to get around this if they want to

**Comment by MeggalBozale** [comment]
Fair idea, I personally would be for that behavior. Though if this is controversial / there will be some discussion needed on this it may be a good idea to open a forums thread, or discussion in imcoder

**Comment by 444Portal** [comment]
Recently been discussing Silicon balance with a few people, both devs and players. I think my stance has shifted to a 'I want this to stay but if it gets merged I wouldn't really care'.

**Comment by Sord213** [comment]
It's definitely a bug, and likely only one that was introduced when the AI eye was turned into an intangible mob and never noticed, considering silicons are not intended to be able to change their own power cells without human intervention.

**Comment by MeggalBozale** [comment]
Thread has been open for 3 days, and reactions seem to be more positive than negative. This is ready to merge unless there's any final objections

**Pull request merged**